yasa/hypno.py

```diff
@@ -474,3 +477,231 @@ def hypno_find_periods(hypno, sf_hypno, threshold="5min", equal_length=False):
 
     new_seq = pd.DataFrame(new_seq)
     return new_seq
+
+
+#############################################################################
+# STAGE CONVERSION
+#############################################################################
+
+
+def hypno_consolidate_stages(hypno, n_stages_in, n_stages_out):
+    """Reduce the number of stages in a hypnogram to match actigraphy or wearables.
+
+    For example, a standard 5-stage hypnogram (W, N1, N2, N3, REM) could be consolidated
+    to a hypnogram more common with actigraphy (W, Light, Deep, REM).
+
+    Parameters
+    ----------
+    hypno : array_like
+        The sleep staging (hypnogram) 1D array.
+    n_stages_in : int
+        Number of possible stages of ``hypno``, where:
+
+        - 5 stages - 0=Wake, 1=N1, 2=N2, 3=N3, 4=REM
+        - 4 stages - 0=Wake, 1=Light, 2=Deep, 3=REM
+        - 3 stages - 0=Wake, 1=NREM, 2=REM
+        - 2 stages - 0=Wake, 1=Sleep
+
+        .. note:: The default YASA values for Unscored (-2) and Artefact (-1) are always allowed.
+    n_stages_out : int
+        Similar to ``n_stages_in`` but for output. Must be higher than ``n_stages_out``.
+
+    Returns
+    -------
+    hypno : array_like
+        The hypnogram, with stages converted to ``n_stages_out`` staging scheme.
+    """
+    assert isinstance(hypno, (list, np.ndarray, pd.Series)), "hypno must be array_like"
+    hypno = np.asarray(hypno, dtype=int).copy()
+    assert n_stages_in in [3, 4, 5], "n_stages_in must be 3, 4, or 5"
+    assert n_stages_out in [2, 3, 4], "n_stages_in must be 2, 3, or 4"
+    assert n_stages_out < n_stages_in, "n_stages_out must be lower than n_stages_in"
+    assert all(s < n_stages_in for s in hypno), "hypno and n_stages_in are not compatible"
+
+    # Change sleep codes (Wake, Artefact, and Unscored never change)
+    if n_stages_out == 2:
```

Comment:
I agree current approach is a bit clunky. But I couldn't come up with a concise way of doing this. The catch is that the replacements are dependent on _both_ `n_stages_out` and `n_stages_in`. For example, if `n_stages_out = 3`, then 3 is N3 if `n_stages_in = 4`, but REM if `n_stages_in = 5`. So I feel like no matter what it requires lots of checks.

Or did you mean to keep the `if else` structure and just use `pd.Series.replace` within each of them instead of changing with numpy indexing? That's an easy change but doesn't seem to be much clearer to me.

Lmk what you'd prefer or if you have a better idea that I'm missing. This is the last thing then I'll push the revision.

Reply:
@remrama I have actually used a different mapping in the Hypnogram branch: https://github.com/raphaelvallat/yasa/blob/8cc01b8d2c3760338a49f7b0cf92bdf834476ecc/yasa/hypno.py#L24-L35

This mapping may be easier to implement because REM is always 4, DEEP always 3, etc. Let me know what you think.

Reply:
ahhh I see -- Yes I love this! Always restricting to lowest numbers has the potential for so much confusion. This is way better. ✅